src/main/java/backend/RepoID.java

```diff
@@ -0,0 +1,83 @@
+package backend;
+
+import org.eclipse.egit.github.core.RepositoryId;
+
+/**
+ * Represents the ID of a repository. To be used to ensure safe comparison
+ */
+public final class RepoID {
+
+    private final String repoOwner;
+    private final String repoName;
+
+    /**
+     * Constructor
+     * @param repoIDString repoOwner and repoName separated by slash
+     * @throws IllegalArgumentException If repoISString is invalid
+     */
+    public RepoID(String repoIDString) throws IllegalArgumentException {
+        boolean isValidRepoID = RepoID.isWellFormedRepoId(repoIDString);
+        if (isValidRepoID) {
+            String[] repoIDComponents = repoIDString.toLowerCase().split("/");
+            this.repoOwner = repoIDComponents[0];
+            this.repoName = repoIDComponents[1];
+        } else {
+            throw new IllegalArgumentException();
+        }
```

Comment:
Reversing the condition will reduce nesting. This is also what John (from Code Bender) when he said 'don't use else' in his talk.
```java
if (!isValidRepoID) {
    throw new IllegalArgumentException();
}
```